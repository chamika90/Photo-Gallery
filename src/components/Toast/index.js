import React from 'react';
import {View, Text, StyleSheet} from 'react-native';

import {theme} from 'config/theme';

const {colors} = theme;

/*
 * Toast component
 */
const Toast = ({text1, settings}) => {
  return (
    <View style={[styles.container, {backgroundColor: settings.color}]}>
      <Text style={styles.message}>{text1}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    paddingVertical: 8,
    paddingHorizontal: 26,
    borderRadius: 30,
    flexDirection: 'row',
    alignItems: 'center',
  },
  message: {
    color: colors.primaryTextColor,
  },
});

export default Toast;
