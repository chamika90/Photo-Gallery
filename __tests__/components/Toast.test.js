import 'react-native';
import React from 'react';
import {render} from '@testing-library/react-native';

import Toast from 'components/Toast';

describe('Toast', () => {
  it('should match snapshot', async () => {
    const {toJSON} = render(
      <Toast text1={'Downloaded'} settings={{color: 'green'}} />,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should display text as Downloaded', async () => {
    const {getAllByText} = render(
      <Toast text1={'Downloaded'} settings={{color: 'green'}} />,
    );

    expect(getAllByText('Downloaded').length).toBe(1);
  });
});
