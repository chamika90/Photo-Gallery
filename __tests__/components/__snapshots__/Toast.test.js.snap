// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 30,
        "flexDirection": "row",
        "paddingHorizontal": 26,
        "paddingVertical": 8,
      },
      Object {
        "backgroundColor": "green",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
      }
    }
  >
    Downloaded
  </Text>
</View>
`;
